app/controllers/todo_items_controller.rb

```diff
@@ -58,6 +58,22 @@ def update
     end
   end
 
+  def destroy
+    todo_item = TodoItem.find(params[:id])
```

Comment:
First, it would be better to set todo_item = TodoItem.where(id: params[:id]).first, which would not throw an exception if there is not a matching record. And if todo_item.blank? is true, then redirect the user back to the index with a flash message stating that the action was unsuccessful. 

But continuing with this approach, one should add a rescue ActiveRecord::RecordNotFound clause to the method to handle the issue and redirect the user to another page with an appropriate message.